pygmt/tests/test_xyz2grd.py

```diff
@@ -19,40 +19,52 @@ def fixture_ship_data():
     return load_sample_bathymetry()
 
 
-def test_xyz2grd_input_file():
+@pytest.fixture(scope="module", name="expected_grid")
+def fixture_grid_result():
     """
-    Run xyz2grd by passing in a filename.
+    Load the expected xyz2grd grid result.
     """
-    output = xyz2grd(data="@tut_ship.xyz", spacing=5, region=[245, 255, 20, 30])
-    assert isinstance(output, xr.DataArray)
-    assert output.gmt.registration == 0  # Gridline registration
-    assert output.gmt.gtype == 0  # Cartesian type
-    return output
+    return xr.DataArray(
+        data=[
+            [-3651.0608, -3015.214, -2320.1033],
+            [-2546.2512, -1977.8754, -963.23303],
+            [-352.3795, -1025.4508, np.nan],
+        ],
+        coords=dict(
+            x=[245.0, 250.0, 255.0],
+            y=[
+                20.0,
+                25.0,
+                30.0,
+            ],
+        ),
+        dims=["y", "x"],
+    )
 
 
-def test_xyz2grd_input_array(ship_data):
+def test_xyz2grd_input_array(ship_data, expected_grid):
```

Comment:
The tests `test_xyz2grd_input_array` and `test_xyz2grd_input_df` can be combined into a single test by using the `pytest.mark.parametrize` decorator, similar to https://github.com/GenericMappingTools/pygmt/blob/7061dacfa227ba12a2f6eaa23df928c556928486/pygmt/tests/test_nearneighbor.py#L24

Reply:
added in d4be06a